### PR TITLE
[8.0] fix: remove warning messages when initializing ElasticDB parameters

### PR DESCRIPTION
**Pull request by aldbr**
From the latest hackaton (16th May 2022).
Remove the warning logs related to ElasticDB initialization, appearing in every job output. 

BEGINRELEASENOTES
*ConfigurationSystem
FIX: remove warning logs when initializing ElasticDB parameters
ENDRELEASENOTES

### CODE REVIEW

File: src/DIRAC/ConfigurationSystem/Client/Utilities.py [hunk 1]
```diff
@@ -462,7 +462,6 @@ def getElasticDBParameters(fullname):
         # No host name found, try at the common place
         result = gConfig.getOption("/Systems/NoSQLDatabases/Host")
         if not result["OK"]:
-            gLogger.warn("Failed to get the configuration parameter: Host. Using localhost")
```
Comment (fstagni) [inline src/DIRAC/ConfigurationSystem/Client/Utilities.py:465]: I am thinking that actually removing them altogether is a bit too aggressive. Maybe we keep them, but adding a boolean flag in the call of the method?

Comment (aldbr) [reply src/DIRAC/ConfigurationSystem/Client/Utilities.py:465]: Do you just want a boolean flag in the function that we set manually if we need? Or do you also want a mechanism to call it with/without the flag from a service/agent? (e.g. a CS parameter)

**Comment by aldbr** [comment]
@fstagni I have some questions.

As a reminder:
Logs appear in the job outputs because `Workflow/ModuleBase` is using `DataManager->StorageElement->DataOperationSender->MonitoringReporter->ServerUtils.getMonitoringDB`.
And the latter intends to access the DB directly, if it fails it generates the logs, and then it uses the client.

Logs are actually generated because `MonitoringDB` inherits from `ElasticDB`, which tries to get parameters from the CS with `Utilities.getElasticDBParameters()`.


- I guess it makes sense to display the log messages when we try to directly connect to the DB.
Even if I set a boolean in this method (`getElasticDBParameters`), it has to be passed in many different classes to be enabled/disabled, isn't it too heavy?
In this case, in which context should we enable it or disable it?

- Another solution would be to prevent `MonitoringReporter` to try accessing directly the DB when used from a job. 
But it is probably too much effort for such a small detail.

- `Utilities.getDBParameters` is similar, but no log is displayed when a parameter is not available.

Wouldn't it be much simpler to either:
- delete these logs? You think it is too agressive so probably no (but at the same time we already do it for DB params...)
- keep these logs? Then we have them in the job outputs, is it a problem?

**Comment by fstagni** [comment]
We are interested in seeing these logs if we are running server side. Maybe we can identify this case using an environment variable? Or a CS location?

**Comment by aldbr** [comment]
> We are interested in seeing these logs if we are running server side. Maybe we can identify this case using an environment variable? Or a CS location?

Ok, so what about using `gConfig.useServerCertificate()` in `ServerUtils`?

**Comment by aldbr** [comment]
> This can be false yet you can be on a server. That's true for any agent downloading proxies.

Ah yes indeed.
Then what would be the most accurate way to check whether we are on a server?

In previous installations, there was the following option in the CS: `InstallType = server`.
It is not present in the Python3 documentation but it seems that it is magically added by DIRAC at the end of the installation.
Could it be used?

If not, what about checking the presence of an option such as `RootPath`, or `Security/CertFile`? Would it be enough?

**Comment by fstagni** [comment]
> In previous installations, there was the following option in the CS: `InstallType = server`. It is not present in the Python3 documentation but it seems that it is magically added by DIRAC at the end of the installation. Could it be used?

I don't think it can be used

> If not, what about checking the presence of an option such as `RootPath`, or `Security/CertFile`? Would it be enough?

`Security/CertFile` can also be used client side, and in general I would not bound (more) things to certificates.
I am not sure why you say to use `RootPath`...?

Inelegantly, maybe the presence of the `runit` directory can be used.

**Comment by aldbr** [comment]
> I don't think it can be used

Any idea why? It could have been practical in this case.

> I am not sure why you say to use RootPath...?
> Inelegantly, maybe the presence of the runit directory can be used.

Well, I should have said `TargetPath` actually.
And indeed, it could be combined to a directory such as `runit`, but it would be inelegant.
Would the `runit` directory be present if kubernetes is used for instance?

What about trying to access the `<DB>` section in the configuration? (e.g. `/Systems/Monitoring/<Setup>/Databases/MonitoringDB`)
If present, we assume that parameters are there, which seems to indicate that the DB can be directly accessed and that we are on the server.
Else, then there is no need to try accessing the DB.

**Comment by fstagni** [comment]
> > I don't think it can be used
> 
> Any idea why? It could have been practical in this case.
> 

I don't see it set anywhere.

> > I am not sure why you say to use RootPath...?
> > Inelegantly, maybe the presence of the runit directory can be used.
> 
> Well, I should have said `TargetPath` actually. And indeed, it could be combined to a directory such as `runit`, but it would be inelegant. Would the `runit` directory be present if kubernetes is used for instance?
> 

Mmm good point. Probably also not an option then. 
And why `TargetPath`?

> What about trying to access the `<DB>` section in the configuration? (e.g. `/Systems/Monitoring/<Setup>/Databases/MonitoringDB`) If present, we assume that parameters are there, which seems to indicate that the DB can be directly accessed and that we are on the server. Else, then there is no need to try accessing the DB.

That is probably better!

**Comment by chaen** [comment]
> That is probably better!

The DB section can be set in the central CS so not an option either

**Comment by chrisburr** [comment]
Converting to draft as I think the solution hasn't been figured out yet

**Comment by aldbr** [comment]
- The `getDBorClient()` function was defined in `MonitoringSystem` and `WorkloadManagementSystem`. Therefore, I moved it in `Core/Utilities`. Let me know whether it should be moved elsewhere.
- Instead of trying to determine whether we are on the server with a flag, I just defined the `Username` and `Password` DB parameters as mandatory. They are checked before any optional parameter, and if they are not present, we return an error. What do you think? 

EDIT: oops, seems like we can have noSQL dbs with no password (integration tests). Thus, either we enforce the need for a password for noSQL databases, or we create a proper flag to know whether we are on a DIRAC server.

**Comment by fstagni** [comment]
> * The `getDBorClient()` function was defined in `MonitoringSystem` and `WorkloadManagementSystem`. Therefore, I moved it in `Core/Utilities`. Let me know whether it should be moved elsewhere.
OK!

> * Instead of trying to determine whether we are on the server with a flag, I just defined the `Username` and `Password` DB parameters as mandatory. They are checked before any optional parameter, and if they are not present, we return an error. What do you think?
> 
Good idea.

> EDIT: oops, seems like we can have noSQL dbs with no password (integration tests). Thus, either we enforce the need for a password for noSQL databases, or we create a proper flag to know whether we are on a DIRAC server.

I think we can enforce the need of user/password.

**Pull request merged**